processing/src/main/java/org/apache/druid/segment/column/ColumnSignature.java

```diff
@@ -50,7 +50,7 @@ class ColumnSignature
 
     // Name must be nonnull, but type can be null (if the type is unknown)
     if (name == null || name.isEmpty()) {
-      throw new IAE(name, "Column name must be non-empty");
+      throw new IAE("Column name must be provided and non-empty");
```

Comment:
Drive by comment, we could've interpolated the name value here:

```
"Column name [%s] must be provided and non-empty", name
```

Would help differentiate between empty and null.  That said, I wonder if this should be throwing an `InvalidInput` instead of `IAE` now.

Reply:
It should probably have been doing `InvalidInput`. I forgot about those! Somehow!

Reply:
@gianm This leads to a 500 error being thrown if the user hits this error, instead of the 400 (which was the older behavior). Raised a PR yesterday which addresses this and cleans up exception handling of MSQ bits that reside in the broker. (https://github.com/apache/druid/pull/14534)